File: app-api/LibElectronAppApi/Database/Models/Artist.cs
using System.ComponentModel.DataAnnotations.Schema;
using LibSqlite3Orm.Abstract.Orm;

namespace LibElectronAppApi.Database.Models;

public class Artist : NamedEntityWithImage
{
    [NotMapped]
    public Lazy<ISqliteQueryable<Album>> Albums { get; set; }
}

File: app-ui/src/models/demoData/Artist.ts
import { NamedEntityWithImage } from "./NamedEntityWithImage";
import {Album} from "./Album";
import {LazyShim} from "../LazyShim";

export class Artist extends NamedEntityWithImage {
    public albums: LazyShim<Album[]> = new LazyShim<Album[]>();
}
